dags/analyzer_helper/discord/discord_transform_raw_data.py

```diff
@@ -0,0 +1,192 @@
+                        author=data["replied_user"],
+                        engaged_users=[data["author"]],
+                        type="receiver",
+                    )
+                    transformed_data.append(receiver_interaction)
+                    # print(f"Added reply interaction: {receiver_interaction}")
+
+                if data.get("user_mentions"):
+                    interactions.append(
+                        self.create_interaction(
+                            data=data,
+                            name="mention",
+                            author=data["author"],
+                            engaged_users=data["user_mentions"],
+                            type="emitter",
+                        )
+                    )
+                    for mentioned_user in data["user_mentions"]:
+                        mentioned_user_interaction = self.create_interaction(
+                            data=data,
+                            name="mention",
+                            author=mentioned_user,
+                            engaged_users=[data["author"]],
+                            type="receiver",
+                        )
+                        transformed_data.append(mentioned_user_interaction)
+                        # print(f"Added mention interaction: {mentioned_user_interaction}")
+
+                all_reaction_users = []
+                if data.get("reactions"):
+                    for reaction in data["reactions"]:
+                        parts = reaction.split(",")
+                        if len(parts) > 2:
+                            all_reaction_users.extend(parts[:-1])
+
+                    if all_reaction_users:
+                        interactions.append(
+                            self.create_interaction(
+                                data=data,
+                                name="reaction",
+                                author=data["author"],
+                                engaged_users=all_reaction_users,
+                                type="receiver",
+                            )
+                        )
+                        for user in all_reaction_users:
+                            emitter_interaction = self.create_interaction(
+                                data=data,
+                                name="reaction",
+                                author=user,
+                                engaged_users=[data["author"]],
+                                type="emitter",
+                            )
+                            transformed_data.append(emitter_interaction)
+                            # print(f"Added reaction interaction: {emitter_interaction}")
+                # print(f"Type of interactions: {type(interactions)}")
+                # print(f"Content of interactions: {interactions}")
+                transformed_item = self.create_transformed_item(
+                    data=data, period=period, interactions=interactions
```

Comment:
`period` is not representative of message date. It's the `createdDate` from data